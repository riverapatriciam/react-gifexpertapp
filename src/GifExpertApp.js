import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () => {
    //    //const category = [...categories, 'HunterXHunter'];
    //    setCategories([...categories, 'HunterXHunter']);
    // }



    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} categories={categories} />
            <hr />


            <ol>
                {
                    categories.map( category => (
                        <GifGrid 
                            key={category}
                            category={category} />
                    ))
                }
            </ol>


        </>

    );
}

