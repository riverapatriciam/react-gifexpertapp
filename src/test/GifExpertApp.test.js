import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp.js';

describe('Pruebas en el componente <GifExpertApp />', () => { 

    test('Debe mostrar correctamente el componente', () => { 

        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();

     });

     test('Debe mostrar una lista de categorias', () => { 

        const categories = ['FMA', 'One Punch'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);


      })

 })