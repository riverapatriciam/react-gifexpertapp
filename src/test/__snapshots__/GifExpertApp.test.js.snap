// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Debe mostrar correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    categories={Array []}
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    categories={
      Array [
        "FMA",
        "One Punch",
      ]
    }
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="FMA"
      key="FMA"
    />
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
